### DISCUSSION TITLE
exclude posts from collections using a filter and custom frontmatter item

### DISCUSSION THREAD

**Discussion by btrem** [11ty/eleventy#2312] [Q&A] [answered]
I can exclude items from collections with `eleventyExcludeFromCollections`. Can I add my own exclude frontmatter item? Say I want to exclude posts from my home page on an ad-hoc basis, but include them on other pages. Using something like this:

    ---
    excludeFromHomePage: true
    ---

What do I need to do in .eleventy.js to accomplish this? I tried creating a filter:

```
eleventyConfig.addNunjucksFilter('excludeByFrontMatter', function(collection, frontMatterItem) {
    if (!frontMatterItem) return collection;
    const filteredCollection = collection.filter(item => !item.data.frontMatterItem);
    return filteredCollection;
});
```

It did not have any effect. Either I'm doing the filter wrong, or this is the wrong approach.

**Comment by pdehaan** [comment, marked as answer]
I think you're _really_ close.
I created https://github.com/pdehaan/11ty-2312 with a couple possible approaches.

```js
eleventyConfig.addFilter("excludeByFrontMatter", function (collection=[], frontMatterItem="") {
  if (!frontMatterItem) {
    return collection;
  }
  return collection.filter(item => !item.data[frontMatterItem]);
});
```

You had `!item.data.frontMatterItem` which will look for a literal key in your front matter named `frontMatterItem`.
By switching that to use the array syntax of `!item.data[frontMatterItem]` it will use the argument you pass to your custom filter.

```njk
{% for item in collections.pages | excludeByFrontMatter("excludeFromHomePage") %}
  <p>{{ item.data.title }}</p>
{% endfor %}
```

---

The second option I tried was just creating a custom collection using the `.addCollection()` API in the .eleventy.js config file.

```js
eleventyConfig.addCollection("filtered_pages", function (collectionApi) {
  const pages = collectionApi.getFilteredByTag("pages");
  return [...pages].filter(page => page.data.excludeFromHomePage !== true);
});
```

I had to hardcode the `excludeFromHomePage` here, but 🤷 .
Then you can just loop over the custom collection.

```njk
{% for item in collections.filtered_pages %}
  <p>{{ item.data.title }}</p>
{% endfor %}
```
